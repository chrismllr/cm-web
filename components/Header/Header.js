import React from 'react'

export default () => (
  <div className="Header">
    <div className="container">
      <h1>Chris Miller</h1>
      <h4>Web developer</h4>
    </div>

    <style jsx>{`
      .Header {
        padding: 2rem 0;
      }

      .Header h4 {
        opacity: 0.5;
        letter-spacing: .03rem;
      }
    `}</style>
  </div>
)
