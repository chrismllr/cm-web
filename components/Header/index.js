import React from 'react'
import { FlexContainer, Section } from '../Layout'

export default () => (
  <Section>
    <FlexContainer
      className="Header"
      justifyContent="space-between"
      flexDirection="column"
      alignItems="flex-start"
    >
      <h1>Chris Miller</h1>
      <h4>UI Engineer</h4>

    </FlexContainer>

    <style jsx>{`
      .Header {
        @media screen and (min-width: 768px) {
          flex-direction: row;
          align-items: center;
        }
      }

      h1 {
        font-size: 1.875rem;
      }
    `}</style>
  </Section>
)
