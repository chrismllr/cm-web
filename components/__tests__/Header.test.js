import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Header from '../Header'

describe('(Component) Header', () => {
  it('Renders title and subtitle of page', () => {
    const wrapper = shallow(
      <Header />
    )

    expect(wrapper.find('h1').text()).toEqual('Chris Miller')
    expect(wrapper.find('h4').text()).toEqual('Web Developer')
  })

  it('Renders to snapshot', () => {
    const component = renderer.create(<Header />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
