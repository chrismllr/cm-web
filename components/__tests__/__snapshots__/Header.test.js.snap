// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Header Renders to snapshot 1`] = `
<div
  className="Header"
  data-jsx={2819036016}
>
  <div
    className="sc-bdVaJa gaHYho"
  >
    <h1
      data-jsx={2819036016}
    >
      Chris Miller
    </h1>
    <h4
      data-jsx={2819036016}
    >
      Web developer
    </h4>
  </div>
</div>
`;
